packages/core/integration-tests/test/javascript.js

```diff
@@ -2955,6 +2955,26 @@ describe('javascript', function () {
     assert.equal(output, 'barbaz');
   });
 
+  it('should insert environment variables from a file even if entry file is specified with source value in package.json', async function () {
+    let b = await bundle(
+      path.join(
+        __dirname,
+        '/integration/env-file-with-package-source/index.js',
+      ),
+      {
+        // when the entry file is specified with a source value in package.json and not specified with cli build target value,
+        // the entry file is resolved to the entry root directory in packages/core/core/src/resolveOptions.js as InitialParcelOptions.
+        // So, entry root directory is specified as InitialParcelOptions' entries value.
+        entries: [
+          path.join(__dirname, '/integration/env-file-with-package-source'),
+        ],
+      },
+    );
```

Comment:
You don't need that entries setting if you just pass the directory as an entry directly instead of the index file.
```suggestion
    let b = await bundle(
      path.join(__dirname, '/integration/env-file-with-package-source'),
    );
```

Reply:
I see, I did not know that I can set a directory as entries when I pass a directory as an entry directly. Thank you for helping!